Mathlib/Data/MvPolynomial/Funext.lean

```diff
@@ -0,0 +1,103 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+
+! This file was ported from Lean 3 source module data.mv_polynomial.funext
+! leanprover-community/mathlib commit da01792ca4894d4f3a98d06b6c50455e5ed25da3
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Polynomial.RingDivision
+import Mathlib.Data.MvPolynomial.Rename
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+## Function extensionality for multivariate polynomials
+
+In this file we show that two multivariate polynomials over an infinite integral domain are equal
+if they are equal upon evaluating them on an arbitrary assignment of the variables.
+
+# Main declaration
+
+* `MvPolynomial.funext`: two polynomials `φ ψ : MvPolynomial σ R`
+  over an infinite integral domain `R` are equal if `eval x φ = eval x ψ` for all `x : σ → R`.
+
+-/
+
+
+namespace MvPolynomial
+
+variable {R : Type _} [CommRing R] [IsDomain R] [Infinite R]
+
+set_option maxHeartbeats 1000000 in
+private theorem funext_fin {n : ℕ} {p : MvPolynomial (Fin n) R}
+    (h : ∀ x : Fin n → R, eval x p = 0) : p = 0 := by
+  induction' n with n ih generalizing R
+  · let e := MvPolynomial.isEmptyRingEquiv R (Fin 0)
+    apply e.injective
+    rw [RingEquiv.map_zero]
+    convert h finZeroElim
+    /- Porting note: this branch of the proof closes here in Lean 4!
+    suffices
+      (eval₂_hom (RingHom.id _) (IsEmpty.elim' Fin.isEmpty)) p =
+        (eval finZeroElim : MvPolynomial (Fin 0) R →+* R) p by
+      rw [← this]
+      simp only [coe_eval₂Hom, is_empty_ring_equiv_apply, RingEquiv.trans_apply,
+        aeval_eq_eval₂Hom]
+      congr
+    exact eval₂Hom_congr rfl (Subsingleton.elim _ _) rfl -/
```

Comment:
I'd say we can just drop this. Very important to leave porting notes for any regressions, but when Lean 4 is just better, it's okay to simply clean up.